routers/crudRouter.js

```diff
@@ -36,53 +36,86 @@ exports.getRouter = function (Model) {
+          }
+        })
+      }).catch(err => {
+        MsLogger.error(err)
+        res.sendStatus(500)
       })
-    }).catch(err => {
-      MsLogger.error(err)
-      res.sendStatus(500)
-    })
+    } else {
+      res.sendStatus(400)
+    }
   })
 
   router.post('/', (req, res) => {
-    MongoClient.connect(MONGU_URI).then(connection => {
-      connection.db().collection(Model.collectionName).findOne(idMongoQuery(Model.IdField, req.body[Model.IdField])).then(data => {
-        if (data) {
-          res.status(400)
-          res.send('Object already exists')
-          return
-        }
-        // TODO: validate
-        connection.db.collection(Model.collectionName).insertOne(req.body).then(a => {
-          if (a.insertedCount > 0) {
-            res.sendStatus(200)
+    let validationResult = true
+    if (options.validationMethods.post) {
+      validationResult = options.validationMethods.post(req.params)
+    }
+
+    if (validationResult) {
+      MongoClient.connect(MONGU_URI).then(connection => {
+        connection.db().collection(options.collectionName).findOne(idMongoQuery(options.IdField, req.body[options.IdField])).then(data => {
+          if (data) {
+            res.status(400)
+            res.send('Object already exists')
+            return
           }
+          connection.db.collection(options.collectionName).insertOne(req.body).then(a => {
+            if (a.insertedCount > 0) {
+              res.sendStatus(201)
+            }
+          })
         })
+      }).catch(err => {
+        MsLogger.error(err)
+        res.sendStatus(500)
       })
-    }).catch(err => {
-      MsLogger.error(err)
-      res.sendStatus(500)
-    })
+    } else {
+      res.sendStatus(400)
+    }
   })
 
   router.delete('/:id', (req, res) => {
-    MongoClient.connect(MONGU_URI).then(connection => {
-      connection.db().collection(Model.collectionName).findOneAndDelete(idMongoQuery(Model.IdField, parseInt(req.params.id))).then(dbResponse => {
-        if (dbResponse.ok === 1) {
-          res.sendStatus(200)
-        }
+    console.log(req.params.id)
+    let validationResult = true
+    if (options.validationMethods.delete) {
+      validationResult = options.validationMethods.delete(req.params)
+      console.log(validationResult)
+    }
+
+    if (validationResult) {
+      MongoClient.connect(MONGU_URI).then(connection => {
+        connection.db().collection(options.collectionName).findOneAndDelete(idMongoQuery(options.IdField, parseInt(req.params.id))).then(dbResponse => {
+          if (dbResponse.ok === 1) {
+            res.sendStatus(200)
+          }
+        })
+      }).catch(err => {
+        MsLogger.error(err)
+        res.sendStatus(500)
       })
-    }).catch(err => {
-      MsLogger.error(err)
-      res.sendStatus(500)
-    })
+    } else {
+      res.sendStatus(400)
+    }
   })
 
+  if (options.extraRouters) {
+    for (const extraRouter of options.extraRouters) {
+      router.use('/', extraRouter)
```

Comment:
Route `'/'` is the default. You don't have to put in the string route parameter if you use it. You could just run `router.use(extraRouter)`